tests/Test/Xrefcheck/RedirectDefaultSpec.hs

```diff
@@ -23,7 +23,7 @@ import Xrefcheck.Verify
 
 test_redirectRequests :: TestTree
 test_redirectRequests = testGroup "Redirect response defaults"
-  [ testGroup "Temporary" $ temporaryRedirectTests <$> [302, 303, 307]
+  [ testGroup "Temporary and 304" $ allowedRedirectTests <$> [302, 303, 304, 307]
```

Comment:
I would put it to a separate group. I think, separation into groups should rather be by semantics, not by the behaviour, and semantics of 304 is quite custom, its behaviour just happened to fall into the same group as temporary redirects.